File: app/src/main/java/com/example/doanck/feature/main/MainActivity.java
package com.example.doanck.feature.main;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;

import com.example.doanck.R;
import com.example.doanck.core.utils.StartButtonHandler;

public class MainActivity extends AppCompatActivity {

    private Button btnStart;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnStart = findViewById(R.id.btnStart);

        // Gọi handler tách riêng
        new StartButtonHandler(this, btnStart);
    }
}

File: app/src/main/java/com/example/doanck/MainActivity.kt
package com.example.doanck

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.os.bundleOf

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onStart() {
        super.onStart()
        // App mở lên / từ nền quay lại -> HomeFragment sẽ nhận signal
        supportFragmentManager.setFragmentResult("APP_OPENED", bundleOf())
    }
}
